libraryServiceHomework

entgorbunov opened the pull request.

entgorbunov committed:
exceptions homework

entgorbunov committed:
try with resources homework

entgorbunov committed:
library service homework

psorokin02 commented on src/main/java/com/oop/LibraryService.java:
```diff
@@ -0,0 +1,68 @@
+public class LibraryService {
+    private final List<Book> books;
+    private final List<User> users;
+    private final HashMap<Long, List<Long>> userBooks;
```
Будет лучше сделать HashMap<Long, Set<Long>> userBooks;

HashSet будет быстрее работать на поиск и вставку, а тебе только эти операции и нужны

psorokin02 commented on src/main/java/com/oop/LibraryService.java:
```diff
@@ -0,0 +1,68 @@
+        return availableBooks;
+    }
+
+    public List<Book> getUserBooks(long userId) {
```
Давай во все методы добавим проверку входных параметров

1. если передали неизвестный userId -> выбросить исключение с соответсвующим текстом
2. если передали неизвестный bookId -> выбросить исключение с соответсвующим текстом

entgorbunov committed:
во все методы добавил проверку входных параметров
если передали неизвестный userId -> выбросить исключение с соответсвующим текстом
если передали неизвестный bookId -> выбросить исключение с соответсвующим текстом.

    private final HashMap<Long, Set<Long>> userBooks;
теперь это не arraylist<Long>
